command/command.go

```diff
@@ -9,121 +9,143 @@ import (
 	"strings"
 )
 
-// ----------
-
-// Model ...
-type Model struct {
+// Command ...
+type Command interface {
+	SetDir(dir string) Command
+	SetStdout(stdout io.Writer) Command
+	SetStderr(stdout io.Writer) Command
+	SetEnvs(envs ...string) Command
+	AppendEnvs(envs ...string) Command
+	Args() []string
+	PrintableCommandArgs() string
+	Run() error
+	RunAndReturnTrimmedOutput() (string, error)
+	RunAndReturnExitCode() (int, error)
+	RunAndReturnTrimmedCombinedOutput() (string, error)
+}
+
+// Factory ...
+type Factory func(name string, args ...string) Command
+
+// NewCommand ...
+func NewCommand(name string, args ...string) Command {
```

Comment:
I think the name of this constructor should be NewXYZCommand like NewRealCommand NewDefaultCommand NewOsCommand, NewWhateverCommand

Reply:
Also seeing other constructors below maybe Command is not needed at the end so from the outside you can write:
`command.NewDefault()` and for the rest `command.NewDefaultWithParams()`.

Reply:
We will never probably use the constructor functions directly just in place of the Factory if we want to keep the NewWithXYZ constructors we probably also need to introduce FactoryWithXYZ for them

Reply:
I think in this case `NewCommand` is fine because there's no missing context here that would justify any extra terms (e.g. `Default` or `OS`). `Command` should be clear to anybody in any context, so I would say there's no need for the extra verbosity.

I forgot to raise it, but I also wanted to ask whether I should rename the existing constructors to contain the word `Command`?

Reply:
A note that it will be used as command.NewCommand ,so adding Command here can be considered 'stuttering' which is  not recommended. See https://stackoverflow.com/a/45886055

Reply:
@lpusok How about we move it to either `cli` or `os`?

Reply:
Discussed offline, we settled with the following naming scheme:
* `command.Command` for the interface.
* `command.New` for the constructor.
* No overloads for the constructor/factory.